### ISSUE TITLE
Feature: Install recent JRuby on Ruby VM

### ISSUE THREAD

**Issue by bf4** [travis-ci/travis-ci#5733]
I'd [like to run a recent JRuby release against Rails 5](https://github.com/rails/rails/pull/23499) but per https://docs.travis-ci.com/user/languages/ruby/#Supported-Ruby-Versions-and-RVM only the 1.7 series is pre-installed in the VM. (Even the precompiled binaries still need to be downloaded and installed: http://rubies.travis-ci.org/ubuntu/12.04/x86_64/jruby-9.0.4.0 )

<img width="920" alt="screen shot 2016-03-02 at 9 14 43 pm" src="https://cloud.githubusercontent.com/assets/142914/13483639/dcb4c598-e0bb-11e5-8219-8c336bdec2bf.png">

<img width="995" alt="screen shot 2016-03-02 at 9 16 10 pm" src="https://cloud.githubusercontent.com/assets/142914/13483669/1e0273ce-e0bc-11e5-8d23-6b751abd5035.png">

<img width="977" alt="screen shot 2016-03-02 at 9 17 51 pm" src="https://cloud.githubusercontent.com/assets/142914/13483678/46cdf4d6-e0bc-11e5-9c49-bae3c7b02403.png">

I see that `ls /home/travis/.rvm/gems` has

```plain
cache   ruby-1.8.7-head   ruby-2.1.2@global
jruby-1.7.19-d18   ruby-1.8.7-head@global  ruby-2.1.3
jruby-1.7.19-d18@global    ruby-1.8.7-p374   ruby-2.1.3@global
jruby-1.7.19-d19   ruby-1.8.7-p374@global  ruby-2.1.4
jruby-1.7.19-d19@global    ruby-1.9.2-p330   ruby-2.1.4@global
jruby-9.0.0.0.pre1   ruby-1.9.2-p330@global  ruby-2.1.5
jruby-9.0.0.0.pre1@global  ruby-1.9.3-p551   ruby-2.1.5@global
jruby-9.0.4.0   ruby-1.9.3-p551@global  ruby-2.2.0
jruby-9.0.4.0@global   ruby-2.0.0-p598   ruby-2.2.0@global
ree-1.8.7-2012.02   ruby-2.0.0-p598@global
ree-1.8.7-2012.02@global   ruby-2.1.2
```

but I'm not sure the relationship between gemsets being present and the binary being available on the VM.

**Comment by bf4** [comment]
It appears 9.0.5.0 is now available from the travis precompiled rubies but they are not used. ref: https://github.com/rails/rails/pull/23499#discussion_r55467311

**Comment by BanzaiMan** [comment]
@bf4 Where's the build that shows it's not used? Seem that the Rails PR has had some forced pushes that hides things.

**Comment by bf4** [comment]
@BanzaiMan this is true re: force pushes.. I'm just trying stuff.  Here's one https://travis-ci.org/rails/rails/jobs/114197771

<img width="609" alt="screen shot 2016-03-09 at 3 36 15 am" src="https://cloud.githubusercontent.com/assets/142914/13631054/30e3c7b8-e5a8-11e5-8bbb-ab9a8b6e93c2.png">

**Comment by bf4** [comment]
And my rvm futzing isn't working   https://travis-ci.org/rails/rails/jobs/114732815

<img width="1029" alt="screen shot 2016-03-09 at 3 38 21 am" src="https://cloud.githubusercontent.com/assets/142914/13631104/88687df8-e5a8-11e5-88f5-d9963dece950.png">

<img width="418" alt="screen shot 2016-03-09 at 3 39 59 am" src="https://cloud.githubusercontent.com/assets/142914/13631122/a075d7d8-e5a8-11e5-9e2f-462e830268b5.png">

**Comment by bf4** [comment]
( @BanzaiMan The goal of the PR is to have a recent JRuby that installs super quick, like the ones pre-installed on the VM, or quick download, use binary, no gemsets, recent bundler etc.)

**Comment by bf4** [comment]
(The `--verbose` thing turned out to be something in Rake fixed in `>= 0.11.1` https://github.com/ruby/rake/pull/120 https://github.com/jruby/jruby/issues/3653#issuecomment-195883717 )
